Enhance logging approach of SummaryJavadocCheck and use AST-based approach to get content of inline tag.

I have downloaded the latest cli from: https://checkstyle.sourceforge.io/config_javadoc.html#SummaryJavadoc
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="SummaryJavadocCheck"/>
    </module>
</module>

/var/tmp $ cat Test.java
public class Test {

    /**
     * {@summary Missing period}
     */
    void foo() {
    }

    /**
     * {@summary Missing period
     * and having multiple lines}
     */
    void hoo() {
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:3: Summary of Javadoc is missing an ending period. [SummaryJavadoc]
[ERROR] /var/tmp/Test.java:9: Summary of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 2 errors.
```

**Describe the solution you'd like**
The logging approach is not optimized, it directly logs the violation at the start of javadoc comment, the least improvement that can be done is to log violation at the `@summary` tag.

Expected output-
```bash
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:4: Summary of Javadoc is missing an ending period. [SummaryJavadoc]
[ERROR] /var/tmp/Test.java:10: Summary of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 2 errors.
```

Additionally, the check currently uses regex in the following method-
https://github.com/checkstyle/checkstyle/blob/c9134ce31c4776e4ab6764824bb234bf5495be0e/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L435-L446

This should be replaced by the AST approach which was initially discussed in https://github.com/checkstyle/checkstyle/issues/9993

> Can there be something above the summary tag? In this case difference in violation positions will be clearer.

@strkkk There can be something above the `@summary` tag but it will be redundant to do so as it won't be a part of the summary. 
Eg-
```
    /**
     * This is the first line.
     * {@summary This is the summary of the method.}
     */
    public void method() {

    }
```
Renders as-
![Screenshot from 2021-12-15 10-07-14](https://user-images.githubusercontent.com/73882557/146124196-be9f4dd7-379b-470a-9859-9f2805c1f049.png)

and Javadoc is generated with a warnnig-
```
/home/vyom/IdeaProjects/ConfigCheckstyle/src/TestP/Test.java:7: warning: invalid use of @summary
     * {@summary This is the summary of the method.}
       ^
```
Though you can see it in module details-
![Screenshot from 2021-12-15 10-11-16](https://user-images.githubusercontent.com/73882557/146124473-91978ae6-4baf-4d82-a230-35903da281ae.png)

> But it seems to me that it is an enhancement for check

Yes, this is an enhancement.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
@@ -22,11 +22,9 @@ package com.puppycrawl.tools.checkstyle.checks.javadoc;
 import java.util.Arrays;
 import java.util.Optional;
 import java.util.Set;
-import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import com.puppycrawl.tools.checkstyle.StatelessCheck;
-import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.api.DetailNode;
 import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
 import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
@@ -251,10 +249,6 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
     private static final Pattern HTML_ELEMENTS =
             Pattern.compile("<[^>]*>");
 
-    /**
-     * This regexp is used to extract the content of a summary javadoc tag.
-     */
-    private static final Pattern SUMMARY_PATTERN = Pattern.compile("\\{@summary ([\\S\\s]+)}");
     /** Period literal. */
     private static final String PERIOD = ".";
 
@@ -309,8 +303,9 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        if (containsSummaryTag(ast)) {
-            validateSummaryTag(ast);
+        final Optional<DetailNode> inlineSummaryTag = getInlineSummaryTag(ast);
+        if (inlineSummaryTag.isPresent()) {
+            validateSummaryTag(inlineSummaryTag.get());
         }
         else if (!startsWithInheritDoc(ast)) {
             final String summaryDoc = getSummarySentence(ast);
@@ -332,18 +327,22 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
     }
 
     /**
-     * Checks if summary tag present.
+     * Gets the inline summary tag.
      *
      * @param javadoc javadoc root node.
-     * @return {@code true} if first sentence contains @summary tag.
+     * @return an optional of inline summary tag node, empty optional if inline tag is not
+     *         a summary tag.
      */
-    private static boolean containsSummaryTag(DetailNode javadoc) {
-        final Optional<DetailNode> node = Arrays.stream(javadoc.getChildren())
+    private static Optional<DetailNode> getInlineSummaryTag(DetailNode javadoc) {
+        Optional<DetailNode> node = Arrays.stream(javadoc.getChildren())
                 .filter(SummaryJavadocCheck::isInlineTagPresent)
                 .findFirst()
                 .map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement);
 
-        return node.isPresent() && isSummaryTag(node.get());
+        if (node.isPresent() && !isSummaryTag(node.get())) {
+            node = Optional.empty();
+        }
+        return node;
     }
 
     /**
@@ -387,13 +386,13 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
     }
 
     /**
-     * Checks if the first tag inside ast is summary tag.
+     * Checks if the javadoc inline tag is {@code {@summary}} tag.
      *
-     * @param javadoc root node.
-     * @return {@code true} if first tag is summary tag.
+     * @param javadocInlineTag node of type {@link JavadocTokenTypes#JAVADOC_INLINE_TAG}
+     * @return {@code true} if inline tag is summary tag.
      */
-    private static boolean isSummaryTag(DetailNode javadoc) {
-        final DetailNode[] child = javadoc.getChildren();
+    private static boolean isSummaryTag(DetailNode javadocInlineTag) {
+        final DetailNode[] child = javadocInlineTag.getChildren();
 
         // Checking size of ast is not required, since ast contains
         // children of Inline Tag, as at least 2 children will be present which are
@@ -405,39 +404,63 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
     /**
      * Checks the inline summary (if present) for {@code period} at end and forbidden fragments.
      *
-     * @param ast javadoc root node.
+     * @param inlineSummaryTag node of type {@link JavadocTokenTypes#JAVADOC_INLINE_TAG}
      */
-    private void validateSummaryTag(DetailNode ast) {
-        final String inlineSummary = getInlineSummary();
+    private void validateSummaryTag(DetailNode inlineSummaryTag) {
+        final String inlineSummary = getContentOfInlineCustomTag(inlineSummaryTag);
         final String summaryVisible = getVisibleContent(inlineSummary);
         if (summaryVisible.isEmpty()) {
-            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+            log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
         }
         else if (!period.isEmpty()) {
             if (isPeriodAtEnd(summaryVisible, period)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
+                log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
             }
             else if (containsForbiddenFragment(inlineSummary)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+                log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_JAVADOC);
             }
         }
     }
 
     /**
-     * Gets entire content of summary tag.
+     * Gets the content of inline custom tag.
      *
-     * @return summary sentence of javadoc root node.
+     * @param inlineTag inline tag node.
+     * @return String consisting of the content of inline custom tag.
      */
-    private String getInlineSummary() {
-        final DetailAST blockCommentAst = getBlockCommentAst();
-        final String javadocText = blockCommentAst.getFirstChild().getText();
-        final Matcher matcher = SUMMARY_PATTERN.matcher(javadocText);
-        String comment = "";
-        if (matcher.find()) {
-            comment = matcher.group(1);
+    public static String getContentOfInlineCustomTag(DetailNode inlineTag) {
+        final DetailNode[] childrenOfInlineTag = inlineTag.getChildren();
+        final StringBuilder customTagContent = new StringBuilder(256);
+        final int indexOfContentOfSummaryTag = 3;
+        if (childrenOfInlineTag.length != indexOfContentOfSummaryTag) {
+            DetailNode currentNode = childrenOfInlineTag[indexOfContentOfSummaryTag];
+            while (currentNode.getType() != JavadocTokenTypes.JAVADOC_INLINE_TAG_END) {
+                extractInlineTagContent(currentNode, customTagContent);
+                currentNode = JavadocUtil.getNextSibling(currentNode);
+            }
         }
-        return JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN.matcher(comment)
-                .replaceAll("");
+        return customTagContent.toString();
+    }
+
+    /**
+     * Extracts the content of inline custom tag recursively.
+     *
+     * @param node DetailNode
+     * @param customTagContent content of custom tag
+     */
+    private static void extractInlineTagContent(DetailNode node,
+        StringBuilder customTagContent) {
+        final DetailNode[] children = node.getChildren();
+        if (children.length == 0) {
+            customTagContent.append(node.getText());
+        }
+        else {
+            for (DetailNode child : children) {
+                if (child.getType() != JavadocTokenTypes.LEADING_ASTERISK) {
+                    extractInlineTagContent(child, customTagContent);
+                }
+            }
+        }
     }
 
     /**

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
@@ -413,7 +413,7 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
             log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
         }
         else if (!period.isEmpty()) {
-            if (isPeriodAtEnd(summaryVisible, period)) {
+            if (isPeriodNotAtEnd(summaryVisible, period)) {
                 log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
             }
             else if (containsForbiddenFragment(inlineSummary)) {
@@ -475,13 +475,13 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
     }
 
     /**
-     * Checks if the string ends with period.
+     * Checks if the string does not end with period.
      *
      * @param sentence string to check for period at end.
      * @param period string to check within sentence.
-     * @return {@code true} if sentence ends with period.
+     * @return {@code true} if sentence does not end with period.
      */
-    private static boolean isPeriodAtEnd(String sentence, String period) {
+    private static boolean isPeriodNotAtEnd(String sentence, String period) {
         final String summarySentence = sentence.trim();
         return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
     }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
@@ -81,6 +81,7 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
             "142: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "147: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "150: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "155: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
         };
         verifyWithInlineConfigParser(
                 getPath("InputSummaryJavadocIncorrect.java"), expected);
@@ -90,16 +91,16 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
     public void testInlineForbidden() throws Exception {
         final String[] expected = {
             "26: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "32: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "38: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "44: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
-            "49: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "54: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "64: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
-            "86: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
-            "99: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "114: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "120: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "31: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "36: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "41: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
+            "45: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "49: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "59: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
+            "80: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
+            "94: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "108: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "115: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
         };
         verifyWithInlineConfigParser(
                 getPath("InputSummaryJavadocInlineForbidden.java"), expected);
@@ -151,19 +152,19 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
     public void testInlineDefaultConfiguration() throws Exception {
         final String[] expected = {
             "22: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "27: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "32: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "42: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "47: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "60: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "65: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "122: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "127: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
-            "132: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "143: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "161: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "166: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "186: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "26: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "30: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "40: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "44: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "56: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "60: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "116: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "120: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
+            "125: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "136: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
+            "153: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "157: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "179: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
         };
 
         verifyWithInlineConfigParser(
@@ -188,8 +189,8 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
     public void testHtmlFormatSummary() throws Exception {
         final String[] expected = {
             "22: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "37: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "42: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "36: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
+            "41: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
         };
 
         verifyWithInlineConfigParser(

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocHtmlFormat.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocHtmlFormat.java
@@ -18,10 +18,9 @@ public class InputSummaryJavadocHtmlFormat {
      * <p><b>{@summary Normal Javadoc.}</b></p>
      */
     private void foo1() {} // ok
-    // violation below
     /**
      * <p><b><i>{@summary Normal Javadoc}</i></b></p>
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     private void foo2() {}
 
     /**
@@ -33,15 +32,14 @@ public class InputSummaryJavadocHtmlFormat {
      * {@summary <p> .</p>}
      */
     private void foo4() {} // ok
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * <p>{@code Code.}</p>
      */
     private void foo6() {}
-    // violation below
     /**
      * <p>{@summary}</p>
-     */
+     */ // violation above 'Summary javadoc is missing.'
     private void foo11() {}
 
     /**

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocIncorrect.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocIncorrect.java
@@ -20,7 +20,7 @@ class InputSummaryJavadocIncorrect {
      * As of JDK 1.1, replaced by {@link #setBounds(int,int,int,int)}
      */
     void foo3() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * @throws Exception if a problem occurs
      */
@@ -38,12 +38,12 @@ class InputSummaryJavadocIncorrect {
      * Some Javadoc.
      */
     public static final byte NUL = 0;
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * <a href="mailto:vlad@htmlbook.ru"/>
      */
      class InnerInputCorrectJavaDocParagraphCheck {
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
          /**
           * foooo@foooo
           */
@@ -53,18 +53,18 @@ class InputSummaryJavadocIncorrect {
          * Some java@doc.
          */
         public static final byte NUL_2 = 0;
-    // violation below
+    // violation below 'Forbidden summary fragment.'
         /**
          * This method
          * returns some javadoc. Some javadoc.
          */
         boolean emulated() {return false;}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * <a href="mailto:vlad@htmlbook.ru"/>
          */
         void foo2() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * @return the
          * customer ID some javadoc.
@@ -75,7 +75,7 @@ class InputSummaryJavadocIncorrect {
          * As of JDK 1.1, replaced by {@link #setBounds(int,int,int,int)}.
          */
         void foo3() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * @throws Exception if a problem occurs
          */
@@ -89,7 +89,7 @@ class InputSummaryJavadocIncorrect {
          */
         void foo6() {}
     }
-    // violation below
+    // violation below 'Forbidden summary fragment.'
      /**
       * A {@code InnerInputCorrectJavaDocParagraphCheck} is a simple code.
       */
@@ -109,7 +109,7 @@ class InputSummaryJavadocIncorrect {
          * As of JDK 1.1, replaced by {@link #setBounds(int,int,int,int)}.
          */
         void foo3() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * @throws Exception if a problem occurs
          */
@@ -122,31 +122,40 @@ class InputSummaryJavadocIncorrect {
          * An especially short bit of Javadoc.
          */
         void foo6() {}
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
          /**
           * mm{@inheritDoc}
           */
          void foo7() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
          /**
           * {@link #setBounds(int,int,int,int)}
           */
          void foo8() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
          /**
           *
           */
          void foo10() {}
     };
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
     /**
      * M m m m {@inheritDoc}
      */
     void foo7() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /** */
     <T> T foo8(T t) {return null;}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /** */
     String[] foo9() {return null;}
+
+    /**
+     * This is the summary.
+     * {@summary This is NOT summary according to javadoc tool}
+     */
+    // violation 2 lines above 'Summary .* missing an ending period.'
+    // until https://github.com/checkstyle/checkstyle/issues/11213
+    void testMethod() {
+    }
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocIncorrect2.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocIncorrect2.java
@@ -19,7 +19,7 @@ class InputSummaryJavadocIncorrect2 {
      * As of JDK 1.1, replaced by {@link #setBounds(int,int,int,int)}
      */
     void foo3() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * @throws Exception if a problem occurs
      */
@@ -37,12 +37,12 @@ class InputSummaryJavadocIncorrect2 {
      * Some Javadoc.
      */
     public static final byte NUL = 0;
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * <a href="mailto:vlad@htmlbook.ru"/>
      */
      class InnerInputCorrectJavaDocParagraphCheck {
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
          /**
           * foooo@foooo
           */
@@ -58,12 +58,12 @@ class InputSummaryJavadocIncorrect2 {
          * returns some javadoc. Some javadoc.
          */
         boolean emulated() {return false;}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * <a href="mailto:vlad@htmlbook.ru"/>
          */
         void foo2() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * @return the
          * customer ID some javadoc.
@@ -74,7 +74,7 @@ class InputSummaryJavadocIncorrect2 {
          * As of JDK 1.1, replaced by {@link #setBounds(int,int,int,int)}.
          */
         void foo3() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * @throws Exception if a problem occurs
          */
@@ -108,7 +108,7 @@ class InputSummaryJavadocIncorrect2 {
          * As of JDK 1.1, replaced by {@link #setBounds(int,int,int,int)}.
          */
         void foo3() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * @throws Exception if a problem occurs
          */
@@ -121,31 +121,31 @@ class InputSummaryJavadocIncorrect2 {
          * An especially short bit of Javadoc.
          */
         void foo6() {}
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
          /**
           * mm{@inheritDoc}
           */
          void foo7() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
          /**
           * {@link #setBounds(int,int,int,int)}
           */
          void foo8() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
          /**
           *
           */
          void foo10() {}
     };
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
     /**
      * M m m m {@inheritDoc}
      */
     void foo7() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /** */
     <T> T foo8(T t) {return null;}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /** */
     String[] foo9() {return null;}
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocInlineDefault.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocInlineDefault.java
@@ -18,17 +18,15 @@ class InputSummaryJavadocInlineDefault {
      * {@summary A simple correct Javadoc.}
      */
     void foo1() {} // ok
-    // violation below
     /**
      * {@summary This code is wrong }
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo5(){}
-    // violation below
     /**
      * {@summary This code {@see Javadoc} is wrong }
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo6(){}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * {@sometag This code {@see Javadoc} is wrong }
      */
@@ -38,15 +36,13 @@ class InputSummaryJavadocInlineDefault {
      * {@summary <p>This code is right.</p>}
      */
     void foo8(){} // ok
-    // violation below
     /**
      * {@summary As of , replaced by {@link #setBounds(int,int,int,int)}}
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo11() {}
-    // violation below
     /**
      * {@summary {@throws Exception if a problem occurs}}
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo12() throws Exception {}
 
     /** {@summary An especially short bit of Javadoc.} */
@@ -56,15 +52,13 @@ class InputSummaryJavadocInlineDefault {
      * {@summary Some Javadoc.}
      */
     public static final byte NUL = 0; // ok
-    // violation below
     /**
      * {@summary <a href="mailto:vlad@htmlbook.ru"/>}
-     */
+     */ // violation above 'Summary javadoc is missing.'
     class InnerInputCorrectJavaDocParagraphCheck {
-    // violation below
         /**
          * {@summary foooo@foooo}
-         */
+         */ // violation above 'Summary .* missing an ending period.'
         public static final byte NUL = 0;
 
         /**
@@ -118,17 +112,16 @@ class InputSummaryJavadocInlineDefault {
          * from {@link #setBounds(int,int,int,int)}.
          */
         void foo3() {} // ok
-    // violation below
         /**
          * {@summary {}@throws Exception if a problem occurs}
-         */
+         */ // violation above 'Summary .* missing an ending period.'
         void foo4() throws Exception {}
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
         /**
          * mm{@inheritDoc}
          */
         void foo7() {}
-    // violation below
+    // violation below 'Summary javadoc is missing.'
         /**
          * {@link #setBounds(int,int,int,int)}
          */
@@ -139,10 +132,9 @@ class InputSummaryJavadocInlineDefault {
          */
         void foo10() {} // ok
     };
-    // violation below
     /**
      * {@summary M m m m {@inheritDoc}}
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo14() {}
 
     /**{@summary @summary .} */
@@ -157,14 +149,13 @@ class InputSummaryJavadocInlineDefault {
      * {@summary {@input Javadoc}.}
      */
     void foo17(){} // ok
-    // violation below
     /**
      * {@summary}
-     */
+     */ // violation above 'Summary javadoc is missing.'
     void foo22() {}
 
     /** */
-    String[] foo9() {return null;} // violation above
+    String[] foo9() {return null;} // violation above 'Summary javadoc is missing.'
 
     /**
      * {@summary Javadoc {@code code} correct.}
@@ -182,7 +173,8 @@ class InputSummaryJavadocInlineDefault {
     private void invalidInlineJavadocTwo() // ok
     {
     }
-    // violation below
+
+    // violation 2 lines below 'Summary .* missing an ending period.'
     /**
      * {@summary first sentence is normally the summary.
      * Use of html tags:

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocInlineForbidden.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocInlineForbidden.java
@@ -22,45 +22,39 @@ public class InputSummaryJavadocInlineForbidden {
      */
     void foo3() { // ok
     }
-    // violation below
     /**
      * {@summary This code is wrong }
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo5() {
     }
-    // violation below
     /**
      * {@summary This code {@see Javadoc} is wrong }
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo6() {
     }
-    // violation below
     /**
      * {@summary As of , replaced by {@link #setBounds(int, int, int, int)}}
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo11() {
     }
-    // violation below
     /**
      * {@summary This method returns something.}
-     */
+     */ // violation above 'Forbidden summary fragment.'
     public static final byte NUL = 0;
-    // violation below
     /**
      * {@summary <a href="mailto:vlad@htmlbook.ru"/>}
-     */
+     */ // violation above 'Summary javadoc is missing.'
     class InnerInputCorrectJavaDocParagraphCheck {
-    // violation below
         /**
          * {@summary foooo@foooo}
-         */
+         */ // violation above 'Summary .* missing an ending period.'
         public static final byte NUL = 0;
 
         /**
          * {@summary Some java@doc.}
          */
         public static final byte NUL_2 = 0; // ok
-    // violation below
+        // violation 2 lines below 'Forbidden summary fragment.'
         /**
          * {@summary @return the
          * customer ID some javadoc.}
@@ -82,7 +76,7 @@ public class InputSummaryJavadocInlineForbidden {
      */
     InputSummaryJavadocInlineForbidden.InnerInputCorrectJavaDocParagraphCheck anon =
             new InputSummaryJavadocInlineForbidden.InnerInputCorrectJavaDocParagraphCheck() {
-                // violation below
+                // violation below 'First sentence .* missing an ending period.'
                 /**
                  * mm{@inheritDoc}
                  */
@@ -95,7 +89,7 @@ public class InputSummaryJavadocInlineForbidden {
                 void foo10() { // ok
                 }
             };
-    // violation below
+    // violation 2 lines below 'Summary .* missing an ending period.'
     /**
      * {@summary first sentence is normally the summary.
      * Use of html tags:
@@ -110,17 +104,18 @@ public class InputSummaryJavadocInlineForbidden {
     public void validInlineJavadoc()
     {
     }
-    // violation below
     /**
      * {@summary <p> </p>}
-     */
+     */ // violation above 'Summary javadoc is missing.'
     void foo12() {
     }
-    // violation below
+
     /**
      * Sentence starts as a plain text sentence
      * {@summary ... but ends in the summary tag}
      */
+    // violation 2 lines above 'Summary .* missing an ending period.'
+    // until https://github.com/checkstyle/checkstyle/issues/11213
     public class TestClass {}
 
     /**

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocPeriod.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocPeriod.java
@@ -10,12 +10,12 @@ period = _
 package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;
 
 public class InputSummaryJavadocPeriod
-{   // violation below
+{   // violation below 'First sentence .* missing an ending period.'
     /**
      * As of JDK 1.1, replaced by {@link #setBounds(int,int,int,int)}
      */
     void foo3() {}
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
     /**
      * Blabla
      */
@@ -33,9 +33,8 @@ public class InputSummaryJavadocPeriod
      * {@summary An especially short bit of Javadoc_}
      */
     void foo7(){}
-    // violation below
     /**
      * {@summary An especially short bit of Javadoc}
-     */
+     */ // violation above 'Summary .* missing an ending period.'
     void foo8() {}
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocPeriodAtEnd.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocPeriodAtEnd.java
@@ -15,28 +15,28 @@ public class InputSummaryJavadocPeriodAtEnd {
      * JAXB 1.0 only default validation event handler
      */
     public static final byte NUL = 0;
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * @throws Exception if a problem occurs.
      */
     public void foo1() throws Exception {
 
     }
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * @return 1.
      */
     public int foo2(){
         return 1;
     }
-    // violation below
+    // violation below 'Summary javadoc is missing.'
     /**
      * <a href="mailto:vlad@htmlbook.ru"></a>
      */
     public void foo3() {
 
     }
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
     /**
      *  A {@code Foo.  Foo}
      */
@@ -56,7 +56,7 @@ public class InputSummaryJavadocPeriodAtEnd {
     public void foo5(){
 
     }
-    // violation below
+    // violation below 'First sentence .* missing an ending period.'
     /**
      * <p>Sentence without period</p>
      *

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/inputs/package-info.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/inputs/package-info.java
@@ -7,7 +7,7 @@ period = (default).
 
 */
 
-/** */ // violation
+/** */ // violation 'Summary javadoc is missing.'
 @Example
 package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc.inputs;
 

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/package-info.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/package-info.java
@@ -7,5 +7,5 @@ period = (default).
 
 */
 
-/** */ // violation
+/** */ // violation 'Summary javadoc is missing.'
 package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;

